### PR TITLE
Rework documentation of std.format.read : formattedRead

### PR DESCRIPTION
**Pull request by berni44**
I renamed the template parameters `R` and `S` to `Range` and `Args` for consistency with other templates used in `std.format`.

I changed the order of the two overloads, like I did with `format` and `sformat` because I think, for beginners the one without template parameters is better comprehensible. For the same reason I did not use idiomatic D for the first example.

### CODE REVIEW

File: std/format/read.d [hunk 1]
```diff
@@ -20,43 +20,44 @@ import std.format.internal.read;
 import std.traits : isSomeString;
 
 /**
-Reads characters from $(REF_ALTTEXT input range, isInputRange, std,range,primitives)
-`r`, converts them according to `fmt`, and writes them to `args`.
+Reads an input range according to a format string and stores the read
```
Comment (RazvanN7) [inline std/format/read.d:23]: I suggest keeping the link the `isInputRange` function.

File: std/format/read.d [hunk 2]
```diff
@@ -20,43 +20,44 @@ import std.format.internal.read;
...
+values into its arguments.
+
+The second version of `formattedRead` takes the format string as
+template argument. In this case, it is checked for consistency at
+compile-time.
```
Comment (RazvanN7) [inline std/format/read.d:28]: Why not put this information as a comment directly above the second function?

Comment (berni44) [reply std/format/read.d:28]: Is this possible? I'll have a look...

Comment (berni44) [reply std/format/read.d:28]: I can't see, how this works out well - maybe I do not know something I should. With my attempts, the comment is either not shown at all or the two overloads are split and I need to duplicate the whole docs... (And we did it this way with `format`, `sformat` and so on too.)

File: std/format/read.d (hunk 1 above)
Comment (RazvanN7) [reply std/format/read.d:23]: Ok.

File: std/format/read.d (hunk 2 above)
Comment (RazvanN7) [reply std/format/read.d:28]: Hmm... I see. I guess this is fine too.

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**